src/components/form-fields/tags-form-field.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  FormControl,
+  FormField,
+  FormMessage,
+  TagBox,
+  TagBoxHandle,
+  TagBoxProps,
+} from "@/shared/ui";
+import { ReactNode, useEffect, useRef } from "react";
+import { FieldValues } from "react-hook-form";
+import {
+  FormFieldProps,
+  FormItemDescription,
+  FormItemLabel,
+} from "./form-fields-common";
+import { isArray, isPopulatedString } from "@/shared/libs";
+
+export function TagsFormField<T extends FieldValues>({
+  name,
+  control,
+  idPrefix,
+  label,
+  description,
+  patterns,
+  onChange,
+  ...props
+}: FormFieldProps<T> & TagBoxProps) {
+  const inputId = idPrefix ? `${idPrefix}-${name}` : undefined;
```

Comment:
I don't think you need inputId here unless you use it for something like \<datalist\>? But it can be a good practice to have ids for all form fields. We can discuss it later.